sql/src/main/java/io/crate/operation/projectors/ProjectingBatchConsumer.java

```diff
@@ -72,17 +72,28 @@ private ProjectingBatchConsumer(BatchConsumer consumer,
                                     RamAccountingContext ramAccountingContext,
                                     ProjectorFactory projectorFactory) {
         this.consumer = consumer;
-        this.projections = projections;
-        this.jobId = jobId;
-        this.ramAccountingContext = ramAccountingContext;
-        this.projectorFactory = projectorFactory;
+        projectors = new ArrayList<>(projections.size());
+
+        boolean projectorsSupportIndependentScrolling = true;
+        for (Projection projection : projections) {
+            Projector projector = projectorFactory.create(projection, ramAccountingContext, jobId);
+            projectors.add(projector);
+
+            if (!projector.providesIndependentScroll()) {
+                projectorsSupportIndependentScrolling = false;
```

Comment:
I think this way if you have `projections = [x, y, z]` where y provides independend scrolling it will also result in requiresScroll = true; 

(the projections are applied in the same order, so  `z(y(x(source))`. If y provides independend scroll then x and source don't have to and z could rely on y to provide scrolling.